exec/java-exec/src/main/java/org/apache/drill/exec/rpc/control/ControlServer.java

```diff
@@ -84,23 +87,33 @@ public ControlConnection initRemoteConnection(SocketChannel channel) {
       public MessageLite getHandshakeResponse(BitControlHandshake inbound) throws Exception {
 //        logger.debug("Handling handshake from other bit. {}", inbound);
         if (inbound.getRpcVersion() != ControlRpcConfig.RPC_VERSION) {
-          throw new RpcException(String.format("Invalid rpc version.  Expected %d, actual %d.", inbound.getRpcVersion(), ControlRpcConfig.RPC_VERSION));
+          throw new RpcException(String.format("Invalid rpc version.  Expected %d, actual %d.",
+              inbound.getRpcVersion(), ControlRpcConfig.RPC_VERSION));
         }
-        if (!inbound.hasEndpoint() || inbound.getEndpoint().getAddress().isEmpty() || inbound.getEndpoint().getControlPort() < 1) {
-          throw new RpcException(String.format("RPC didn't provide valid counter endpoint information.  Received %s.", inbound.getEndpoint()));
+        if (!inbound.hasEndpoint() ||
+            inbound.getEndpoint().getAddress().isEmpty() ||
+            inbound.getEndpoint().getControlPort() < 1) {
+          throw new RpcException(String.format("RPC didn't provide valid counter endpoint information.  Received %s.",
+                  inbound.getEndpoint()));
         }
         connection.setEndpoint(inbound.getEndpoint());
 
         // add the
         ControlConnectionManager manager = connectionRegistry.getConnectionManager(inbound.getEndpoint());
 
         // update the close handler.
-        proxyCloseHandler.setHandler(manager.getCloseHandlerCreator().getHandler(connection, proxyCloseHandler.getHandler()));
+        proxyCloseHandler.setHandler(manager.getCloseHandlerCreator().getHandler(connection,
+            proxyCloseHandler.getHandler()));
 
         // add to the connection manager.
         manager.addExternalConnection(connection);
 
-        return BitControlHandshake.newBuilder().setRpcVersion(ControlRpcConfig.RPC_VERSION).build();
+        final BitControlHandshake.Builder builder = BitControlHandshake.newBuilder();
+        builder.setRpcVersion(ControlRpcConfig.RPC_VERSION);
+        if (config.getAuthProvider() != null) {
```

Comment:
There is only one authProvider per drillbit, but auth can be enabled user to bit, but not bit to bit, which means relying on being empty is not sufficient.